src/assets/js/fullpostpage/unlock-content/edit-fullpp.js

```diff
@@ -0,0 +1,302 @@
+import { __ } from "@wordpress/i18n";
+import {
+  PanelBody,
+  PanelRow,
+  TextControl,
+  SelectControl,
+  Button,
+  Notice,
+} from "@wordpress/components";
+import { useEffect, useState } from "react";
+import apiFetch from "@wordpress/api-fetch";
+import { useSelect } from '@wordpress/data';
+import "../../../scss/admin/editor.scss";
+
+const EditFullPP = ({ locks, ethereumNetworks, onUpdateLocks, onUpdateEthereumNetworks }) => {
+  
+  const [saveMessage, setSaveMessage] = useState(null);
+  const [refresh, setRefresh] = useState(false); //refresh lock(s) meta panel to keep track of locks already added
+
+
+  //get current post id
+  const getpost = useSelect((select) => {
+    const { getCurrentPostId } = select("core/editor");
+    const id = getCurrentPostId();
+  
+    return {
+      id,
+    };
+  }, []);
+
+  //check if current post is already 
+  const { isSavingPost, didPostSaveRequestSucceed } = useSelect((select) => {
+    const { isSavingPost, didPostSaveRequestSucceed } = select("core/editor");
+    return {
+      isSavingPost: isSavingPost(),
+      didPostSaveRequestSucceed: didPostSaveRequestSucceed(),
+    };
+  }, []);
+
+
+  //fetch locks attributes
+  const fetchLockSettings = async (post_id) => {
+    try {
+      const response = await apiFetch({
+        path: `/unlock-protocol/v1/get_unlockpfullpp_attributes?post_id=${post_id}`,
+      });
+
+      if (response && response.locks) {
+        onUpdateLocks(response.locks);
+      }
+    } catch (error) {
+      console.error("Error fetching lock settings:", error);
+    }
+  };
+
+  //useEffect 1: that fetches the networks
+  useEffect(() => {
+    apiFetch({
+      path: "/unlock-protocol/v1/settings",
+    })
+      .then((resp) => {
+        let networks = resp.networks;
+        let selectOptions = [
+          {
+            label: __("None", "unlock-protocol"),
+            value: -1,
+          },
+        ];
+
+        Object.entries(networks).forEach(([key, item]) => {
+          selectOptions.push({
+            label: item.network_name,
+            value: item.network_id,
+          });
+        });
+
+        onUpdateEthereumNetworks(selectOptions);
+
+        // Fetch lock settings from the post meta after networks are fetched
+        fetchLockSettings(getpost.id);
+      })
+      .catch((err) => { });
+  }, [getpost.id, refresh]); // refresh the useEffect anytime either of this changes
+
+
+  //useEffect 2: called when post/page publish/update button clicked
+  useEffect(() => {
+    if (!isSavingPost && didPostSaveRequestSucceed) {
+
+      // refresh the useEffect 1 to reload meta panel in post/page editor sidebar when publish/update button is clicked
+      setRefresh(!refresh); 
+    }
+  }, [isSavingPost, didPostSaveRequestSucceed, getpost.id]);
+
+
+  // Check if the Ethereum address is valid
+  const isAddressValid = (address) => {
```

Comment:
Maybe move this to a utils file? I suspect we already have a version of this existing in the plugin?

Reply:
This is a check that is needed when the reactjs is transpiled to js in the plugin folder so shouldn't be separated.
Kindly check the block implementation, it was used directly too.
So, this can be considered fixed.